libs/prov/prov_k8s_cortx_deploy.py

```diff
@@ -1293,53 +1293,53 @@ def test_deployment(self, sns_data, sns_parity,
                                                               common_const.RGW_CONTAINER_NAME,
                                                               self.deploy_cfg["rgw_rpm"])
                             assert_utils.assert_true(resp[0], resp[1])
-            if self.deployment_type not in self.exclusive_pod_list:
-                if setup_client_config_flag:
-                    LOGGER.info("Setting the current namespace")
-                    resp_ns = master_node_list[0].execute_cmd(
-                        cmd=common_cmd.KUBECTL_SET_CONTEXT.format(namespace),
-                        read_lines=True)
-                    LOGGER.debug("response is %s,", resp_ns)
-                    resp = system_utils.execute_cmd(
-                        common_cmd.CMD_GET_IP_IFACE.format(self.deploy_cfg['iface']))
-                    eth1_ip = resp[1].strip("'\\n'b'")
+        if self.deployment_type not in self.exclusive_pod_list:
+            if setup_client_config_flag:
+                LOGGER.info("Setting the current namespace")
+                resp_ns = master_node_list[0].execute_cmd(
+                    cmd=common_cmd.KUBECTL_SET_CONTEXT.format(namespace),
+                    read_lines=True)
+                LOGGER.debug("response is %s,", resp_ns)
+                resp = system_utils.execute_cmd(
+                    common_cmd.CMD_GET_IP_IFACE.format(self.deploy_cfg['iface']))
+                eth1_ip = resp[1].strip("'\\n'b'")
+                if self.service_type == "NodePort":
+                    resp = ext_lbconfig_utils.configure_nodeport_lb(master_node_list[0],
+                                                                    self.deploy_cfg['iface'])
+                    if not resp[0]:
+                        LOGGER.debug("Did not get expected response: %s", resp)
+                    ext_ip = resp[1]
+                    port = resp[2]
+                    ext_port_ip = self.deploy_cfg['https_protocol'].format(ext_ip)+\
+                                  ":{}".format(port)
+                    LOGGER.debug("External LB value, ip and port will be: %s", ext_port_ip)
+                else:
+                    LOGGER.info("Configure HAproxy on client")
+                    ext_lbconfig_utils.configure_haproxy_rgwlb(master_node_list[0].hostname,
+                                                               master_node_list[0].username,
+                                                               master_node_list[0].password,
+                                                               eth1_ip, self.deploy_cfg['iface'])
+                    ext_port_ip = self.deploy_cfg['https_protocol'].format(eth1_ip)
+                LOGGER.info("Step to Create S3 account and configure credentials")
+                if self.s3_engine == 2:
```

Comment:
What is the meaning of value "2" here? Some comment would help.

Reply:
2 is denoted for rgw based operations its being coming from setup entry and we are comparing it

Reply:
OK. So kindly add an inline comment as appropriate. Thanks!